src/convenience/frameworks.ts

```diff
@@ -80,20 +80,29 @@ export type LambdaAsyncAdapter = (
     _context: unknown,
 ) => ReqResHandler;
 
-export type AzureAdapter = (request: {
-    body?: unknown;
-}, context: {
+export type AzureAdapter = (context: {
     res?: {
-        status: number;
-        body: string;
-        headers?: Record<string, string>;
-        set?: (key: string, value: string) => void;
-        send?: {
-            (body: unknown): void;
-            (status: number, body: unknown): void;
-        };
+        // deno-lint-ignore no-explicit-any
+        [key: string]: any;
     };
-}) => ReqResHandler;
+}, request: { body?: unknown }) => ReqResHandler;
+type AzureV4HttpRequest = {
+    method: string;
+    url: string;
+    headers: any;
+    query: URLSearchParams;
+    body: any;
+}
+type AzureV4InvocationContext = {
+    invocationId: string;
+    functionName: string;
```

Comment:
In addition, these properties here are defined needlessly. They can be dropped because grammY never accesses them. TypeScript's structural type system will still allow passing in objects that have these properties, so the code will continue to work for library consumers.